presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -607,6 +607,10 @@ private synchronized void checkLockHeld(String message)
 
     private static ListenableFuture<?> firstFinishedFuture(List<ListenableFuture<?>> futures)
     {
+        if (futures.size() == 1) {
+            return futures.get(0);
```

Comment:
I'm not sure if this creates a problem since cancel would now propagate (not sure if any one is calling cancel)

Reply:
Is there a contract for `operator.isBlocked().cancel(..)`? I think anything could happen (including query failure e.g. in case of LJO's `lookupSourceProviderFuture` or `spillInProgress`, because LJO checks the future was completed successfully)

But, is there a contract for `Driver.process().cancel()`? What would that mean?